trl/trainer/callbacks.py

```diff
@@ -293,11 +297,22 @@ def on_evaluate(self, args: TrainingArguments, state: TrainerState, control: Tra
             )
 
             completions = list(zip(self.ref_completions, completions))
-            winner_indices = self.judge.judge(prompts, completions)
+            winner_indices = self.judge.judge(prompts, completions, self.shuffle_order)
+            prompts = gather_object(prompts)
+            completions = gather_object(completions)
             winner_indices = gather_object(winner_indices)
 
         # Logging
         if self.trainer.accelerator.is_main_process:
+            global_step = [str(state.global_step)] * len(prompts)
```

Comment:
Do we now require users to have wandb to use this callback?

Reply:
Yes, we now subclass `WandbCallback` so we can access the `self._wandb.log()` method, similar to `LogCompletionsCallback`.

Let me see if I can refactor it to make this optional (i.e. check if `wandb` is in `args.report_to` and only then log the table.

Reply:
Found a good solution in to avoid requiring `wandb`: https://github.com/huggingface/trl/pull/2116/commits/f142a273d602a42ac01dd6f4b05f8ec9775fcf24